backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/utils/SysMLKeywordSwitch.java

```diff
@@ -0,0 +1,128 @@
+package org.eclipse.syson.sysml.export.utils;
+
+import com.google.common.base.Strings;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.syson.sysml.AttributeDefinition;
+import org.eclipse.syson.sysml.AttributeUsage;
+import org.eclipse.syson.sysml.EnumerationDefinition;
+import org.eclipse.syson.sysml.EnumerationUsage;
+import org.eclipse.syson.sysml.InterfaceDefinition;
+import org.eclipse.syson.sysml.InterfaceUsage;
+import org.eclipse.syson.sysml.ItemDefinition;
+import org.eclipse.syson.sysml.ItemUsage;
+import org.eclipse.syson.sysml.PartDefinition;
+import org.eclipse.syson.sysml.PartUsage;
+import org.eclipse.syson.sysml.PortDefinition;
+import org.eclipse.syson.sysml.PortUsage;
+import org.eclipse.syson.sysml.util.SysmlSwitch;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Keyword provider for SysML textual representation.
+ *
+ * @author Arthur Daussy
+ */
+public class SysMLKeywordSwitch extends SysmlSwitch<String> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SysMLKeywordSwitch.class);
+
+    private static final String PART_KEYWORD = "part";
+
+    private static final String PORT_KEYWORD = "port";
+
+    private static final String INTERFACE_KEYWORD = "interface";
+
+    private static final String ITEM_KEYWORD = "item";
+
+    private static final String ENUM_KEYWORD = "enum";
+
+    private static final String ATTRIBUTE_KEYWORD = "attribute";
+
+    @Override
+    public String caseAttributeDefinition(AttributeDefinition object) {
+        return ATTRIBUTE_KEYWORD;
+    }
+
+    @Override
+    public String caseAttributeUsage(AttributeUsage object) {
+        return ATTRIBUTE_KEYWORD;
+    }
+
+    @Override
+    public String caseEnumerationUsage(EnumerationUsage object) {
+        if (object.getOwningDefinition() instanceof EnumerationDefinition && !Strings.isNullOrEmpty(object.getName())) {
+            // Inside a enumeration definition the keyword enum is optional
+            return "";
+        }
+        return ENUM_KEYWORD;
+    }
+
+    @Override
+    public String caseEnumerationDefinition(EnumerationDefinition object) {
+        return ENUM_KEYWORD;
+    }
+
+    @Override
+    public String caseItemDefinition(ItemDefinition object) {
+        return ITEM_KEYWORD;
+    }
+
+    @Override
+    public String caseItemUsage(ItemUsage object) {
+        return ITEM_KEYWORD;
+    }
+
+    @Override
+    public String caseInterfaceDefinition(InterfaceDefinition object) {
+        return INTERFACE_KEYWORD;
+    }
+
+    @Override
+    public String caseInterfaceUsage(InterfaceUsage object) {
+        return INTERFACE_KEYWORD;
+    }
+
+    @Override
+    public String casePortUsage(PortUsage object) {
+        return PORT_KEYWORD;
+    }
+
+    @Override
+    public String casePortDefinition(PortDefinition object) {
+        return PORT_KEYWORD;
+    }
+
+    @Override
+    public String casePartDefinition(PartDefinition object) {
+        return PART_KEYWORD;
```

Comment:
Shouldn't it be _part def_ instead of just _part_? (same remark for all other definitions)